docx_editor/ooxml/unpack.py

```
@@ -67,26 +75,48 @@ def unpack_document(input_file: str | Path, output_dir: str | Path) -> str:
-    xml_files = list(output_path.rglob("*.xml")) + list(output_path.rglob("*.rels"))
-    for xml_file in xml_files:
-        content = xml_file.read_text(encoding="utf-8")
-        dom = defusedxml.minidom.parseString(content)
-        xml_file.write_bytes(dom.toprettyxml(indent="  ", encoding="utf-8"))
+        # Validate ZIP entries before extractall (and before mkdir) so a
+        # rejection leaves the filesystem untouched.
+        try:
+            with zipfile.ZipFile(input_path) as zf:
+                for info in zf.infolist():
+                    if _is_unsafe_zip_path(info.filename):
+                        raise InvalidDocumentError(f"Unsafe ZIP entry path: {info.filename!r} in {input_file}")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)